test/functional/readpreference.test.js

```diff
@@ -759,4 +759,35 @@ describe('ReadPreference', function() {
         });
     })
   });
+
+  it('should use session readPreference instead of client readPreference', {
+    metadata: { requires: { topology: ['single', 'replicaset'] } },
+    test: function(done) {
+      const configuration = this.configuration;
+      const client = this.configuration.newClient(configuration.writeConcernMax(), {
+        useUnifiedTopology: true,
```

Comment:
In general we try to avoid forcing the unified topology like this, and instead filter based on it. You can do this via the test metadata: `requires: { unifiedTopology: false, topology: ... }`.